pkg/native/serializers/serializer_cdx.go

```diff
@@ -45,6 +47,10 @@ func (s *CDX) Serialize(bom *sbom.Document, _ *native.SerializeOptions, _ interf
 
 	doc := cdx.NewBOM()
 	doc.SerialNumber = bom.Metadata.Id
+	if doc.SerialNumber == "" || !isValidCycloneDXSerialNumber(doc.SerialNumber) {
+		doc.SerialNumber = "urn:uuid:" + uuid.New().String()
```

Comment:
Lets use here a deterministic algorithm to generate the new serial number. When the protobom document has an ID defined, use an uuid sha1 to generate the uuid, something like:

```golang
uuid.NewSHA1(uuid.MustParse(`5dbcd03c-dd56-4fff-97af-77f89c66eeba`), []byte(protobom id))
```